PracticeTestsBackend/app/core/main.py

```
@@ -0,0 +1,53 @@
+load_dotenv()
+
+PDF_DIR = os.environ.get("PDF_DIR", "data/input_pdfs")
+TEXT_DIR = os.environ.get("TEXT_DIR", "data/extracted_text")
+DIAGRAMS_ROOT = os.environ.get("DIAGRAMS_ROOT", "data/diagrams")
+
+
+def process_pdf(pdf_path):
+    print(f"[DEBUG] → extracting from: {pdf_path}", flush=True)
+    base       = os.path.splitext(os.path.basename(pdf_path))[0]
+    txt_path   = os.path.join(TEXT_DIR, f"{base}.txt")
+    json_path  = os.path.join(TEXT_DIR, f"{base}.json")
```

Ruff (0.12.2)

Local variable `json_path` is assigned to but never used

Remove assignment to unused variable `json_path`

(F841)